modules/openapi-generator/src/test/java/org/openapitools/codegen/java/spring/SpringCodegenTest.java

```diff
@@ -4882,7 +4882,7 @@ public void testCollectionTypesWithDefaults_issue_18102() throws IOException {
         codegen.additionalProperties().put(CodegenConstants.API_NAME_SUFFIX, "Controller");
         codegen.additionalProperties().put(CodegenConstants.API_PACKAGE, "xyz.controller");
         codegen.additionalProperties().put(CodegenConstants.MODEL_NAME_SUFFIX, "Dto");
-        codegen.setContainerDefaultToNull(true);
+        codegen.setContainerDefaultToNull("true");
```

Comment:
For backward compatibility. I would add two methods the old one with boolean and new one with string

```
void setContainerDefaultToNull(boolean b){
setContainerDefaultToNull(b.toString())
}

void setContainerDefaultToNull(String b){
...
}
```